crimsobot/cogs/utilities.py

```diff
@@ -14,6 +18,113 @@ class Utilities(commands.Cog):
     def __init__(self, bot: CrimsoBOT):
         self.bot = bot
 
+    @commands.command(brief='Create a poll!')
+    async def poll(self, ctx: commands.Context, *, input: str) -> None:
+        """Make a poll! Use the form of some question;option 1;option 2;etc.
+        For example:
+        
+        >poll What should I eat?;tacos;ramen;the rich
+
+        Polls can have up to 20 choices, but keep in mind Discord's 2000-character message limit.
+        Polls technically never close, but you can >tally the results whenever you like!
+        """
+
+        poll_pool = [
+            '1️⃣', '2️⃣', '3️⃣', '4️⃣', '5️⃣', '6️⃣', '7️⃣', '8️⃣', '9️⃣', '🔟',
+            '🇦', '🇧', '🇨', '🇩', '🇪', '🇫', '🇬', '🇭', '🇮', '🇯'
+        ]
+
+        choices = input.split(';')
+        question = choices.pop(0)
+
+        poll_body = ['**{}**\n'.format(question.strip())]
+        for idx, choice in enumerate(choices):
+            poll_body.append('{} {}'.format(poll_pool[idx], choice.strip()))
+
+        embed = c.crimbed(
+            title='{} has created a poll!'.format(ctx.author),
+            descr='{}'.format('\n'.join(poll_body)),
+            thumb_name='think',
+            footer='\u200dPoll ID: {d.month}{d.day}{d.hour}{d.minute}{d.second}'.format(d=datetime.utcnow())
+        )
+
+        msg = await ctx.send(embed=embed)
+
+        # add reactions to msg
+        for emoji in poll_pool[0:len(choices)]:
+            try:
+                await msg.add_reaction(emoji)
+                await asyncio.sleep(0.36)  # smoother rollout of reactions
+            except Exception:
+                await ctx.send('**Someone added emojis!** Wait for me to add them, then choose. `Poll crashed.`')
+                return
+
+    @commands.command(aliases=['tally'], brief='Tally results of a poll!')
+    async def polltally(self, ctx: commands.Context, poll_id: Optional[str] = None) -> None:
+        """Tally the results of the most recent poll in a channel, or of a previous poll if you give the poll's ID.
+        You can only tally the results of a poll in the same channel as the poll.
+        Tallying a poll does not close the poll; its results can be tallied again if it's not too old."""
+
+        # so this is kinda hacky...
+        # first, check channel's message history (backwards from present) for prior polls from bot
+        try:
+            async for message in ctx.channel.history(limit=10000):
```

Comment:
This is potentially performing 100 requests to the history endpoint per invocation. (100 pages of 100 messages from the API, each "page" is another request.)

Maybe tone this down a bit? I can imagine lots of ratelimits (and potentially cloudflare bans) if the command gets spammed.

Reply:
ugh really?

Reply:
Well fffff this will have to do for now